payslip.sh

```diff
@@ -183,6 +183,10 @@ TOKEN="$(wget ${VERBOSE} -O - \
   'https://fress2.adp.com/core/coreControl.asp?ProductType=0' \
   | grep sessionToken | cut -d "'" -f2)"
 
+if [ ${#TOKEN} = 0 ]; then
```

Comment:
why not 
`[ -z "${TOKEN}" ] && fail 'unable to get session token'`

Reply:
I've no preference between `[ ${#token} = 0 ]` and `[ -z "${TOKEN}" ]`; I'm equally unfamiliar with either. Is there a reason to prefer one over the other?

Reply:
I don't like using `&&` as a way of controlling execution. I think it is less clear than using the `if`.

Reply:
I guess in both cases the reasoning is based around the normal practice.  Length checking is more obscure than performing the empty test. Similarly, for one-line check-and-exit tests it is common practice to use the && for execution control in shell. I don't find it harder to parse, but I don't have as strong feelings on this as many will.

Reply:
changed to empty test. not adopting the idiomatic use of `&&`.